xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-ui/src/main/resources/XWiki/AttachmentSelector.xml

```diff
@@ -637,7 +637,8 @@ $xwiki.jsx.use($attachmentPickerDocName)
     /** AJAX deletion of attachments. */
     onDelete : function(event) {
       event.stop();
-      deleteTool = event.element();
+      let deleteTool = event.element();
+      if (!deleteTool.hasClassName('btn')) deleteTool = deleteTool.up('.btn');
```

Comment:
I'd try to use the native API:


```suggestion
      const deleteTool = event.target.closest('.btn');
```

See https://developer.mozilla.org/en-US/docs/Web/API/Event/target and https://developer.mozilla.org/en-US/docs/Web/API/Element/closest (note that ``closest`` starts from the element is being called on).